src/pages/en/reference/api-reference.md

```diff
@@ -158,6 +158,51 @@ You could pass a callback function that renders our the message:
 <div>Hello world!</div>
 ```
 
+### `Astro.self`
+
+within the component template, allowing you to easily iterate over large data stores and nested data-structures.
+
+```astro
+---
+// NestedHeadings.astro
+const { headings } = Astro.props;
+---
+<ul class="nested-headings">
+  <li>{headings.map((item) => {
+    if (Array.isArray(item)) {
+      // If there is a nested data-structure we render `<Astro.self>`
+      // and can pass props through with the recursive call
+      return <Astro.self headings={item} />;
+    } else {
+      return item;
+    }
+  })}</li>
+</ul>
+```
+
+This component could then be used like this:
+
+```astro
+---
+import NestedHeadings from './NestedHeadings.astro';
+---
+<NestedHeadings headings={['A', ['B', 'C'], 'D']} />
+```
+
+And would render HTML like this:
+
+```html
+<ul class="nested-headings">
+  <li>A</li>
+  <li>
+    <ul class="nested-headings">
+      <li>B</li>
+      <li>C</li>
+    </ul>
+  </li>
+  <li>D</li>
+</ul>
+```
```

Comment:
Sorry for vacillating here @aFuzzyBear. I think maybe having this just be “list” instead of the more specific headings might be clearer?

~~~suggestion
```astro
---
// NestedList.astro
const { items } = Astro.props;
---
<ul class="nested-list">
  <li>{items.map((item) => {
    if (Array.isArray(item)) {
      // If there is a nested data-structure we render `<Astro.self>`
      // and can pass props through with the recursive call
      return <Astro.self items={item} />;
    } else {
      return item;
    }
  })}</li>
</ul>
```

This component could then be used like this:

```astro
---
import NestedList from './NestedList.astro';
---
<NestedList items={['A', ['B', 'C'], 'D']} />
```

And would render HTML like this:

```html
<ul class="nested-list">
  <li>A</li>
  <li>
    <ul class="nested-list">
      <li>B</li>
      <li>C</li>
    </ul>
  </li>
  <li>D</li>
</ul>
```
~~~

Reply:
no problem in the slightest, list was my first thought, but thoughts of, perhaps a usable example, but given he code demo, list feels like it would suit the generic format you've provided for us dude